src/components/TextEditor/TextEditor.js

```diff
@@ -0,0 +1,61 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faPalette } from '@fortawesome/free-solid-svg-icons'
+import { faBoxArchive } from '@fortawesome/free-solid-svg-icons'
+import { faThumbTack } from '@fortawesome/free-solid-svg-icons'
+import React from 'react'
+import './TextEditor.css'
+import { useState } from 'react'
+import { StateContext } from '../../Context/StateProvider'
+import ColorPicker from '../ColorPicker/ColorPicker'
+import { useContext } from 'react'
+import { AuthContext } from '../../Context/AuthProvider'
+
+const TextEditor = () => {
+  const { state, dispatch } = useContext(StateContext)
+  const { token } = useContext(AuthContext)
+  const [title, setTitle] = useState('');
+  const [desc, setDesc] = useState('')
+  const [color, setColor] = useState('')
+  const [isPinned, setIsPinned] = useState(false)
+  const [label, setLabel] = useState('')
+  const noteObj = {
+    title, desc, color, isPinned, label
```

Comment:
Suggestion - instead of creating separate states and then adding them to an object. You could create a state object and add these as keys.